go-crud/books/controllers.go

```diff
@@ -0,0 +1,257 @@
+package books
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"os"
+	"strconv"
+	"strings"
+)
+
+var data []book
+
+func loadBookFile() error {
+	_, err := os.Stat("books.json")
+
+	if os.IsNotExist(err) {
+		str, err := json.Marshal(data)
+
+		if err != nil {
+			log.Println(err)
+			return err
+		}
+		err = ioutil.WriteFile("books.json", str, 0644)
+		if err != nil {
+			log.Println(err)
+			return err
+		}
+	}
+
+	rawData, err := ioutil.ReadFile("books.json")
+
+	if err != nil {
+		log.Println(err)
+		return err
+	}
+
+	// fmt.Print("data:  ", string(rawData))
+	err = json.Unmarshal(rawData, &data)
+	if string(rawData) == "null" {
+		data = make([]book, 0)
+	}
+	if err != nil {
+		log.Println(err)
+		return err
+	}
+
+	return nil
+}
+
+func createBook(w http.ResponseWriter, r *http.Request) {
+	if r.Method == "POST" {
+		err := loadBookFile()
+
+		if err != nil {
+			log.Println(err.Error())
```

Comment:
See, you are handling the error you also handled above. Keep this one and omit the logging in the method.